From 5136b8eb33bdd7e1ca9f9a3e649cb11ff979c350 Mon Sep 17 00:00:00 2001
From: Andrew Hou <Andrew.Hou@emc.com>
Date: Thu, 14 Jan 2016 11:50:48 +0800
Subject: [PATCH] Fix compiling failure on Ubuntu 14.04

1. Disable default devel config that has strict gcc 4.8.1 check on Ubuntu 14.04.3
2. Fix new ext2_fs.h location on Ubuntu 14.04.3
3. Apply new mingw64 tool on Ubuntu 14.04.3
---
 MCONFIG                   | 2 +-
 libinstaller/linuxioctl.h | 2 +-
 win64/find-mingw64.sh     | 1 +
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/MCONFIG b/MCONFIG
index a71fd13..b5adefa 100644
--- a/MCONFIG
+++ b/MCONFIG
@@ -77,7 +77,7 @@ UMAKEDEPS = -Wp,-MT,$@,-MMD,$(dir $@).$(notdir $@).d

 # Items that are only appropriate during development; this file is
 # removed when tarballs are generated.
--include $(topdir)/MCONFIG.devel
+#-include $(topdir)/MCONFIG.devel

 # Local additions, like -DDEBUG can go here
 -include $(topdir)/MCONFIG.local
diff --git a/libinstaller/linuxioctl.h b/libinstaller/linuxioctl.h
index 7ef919a..100fd86 100644
--- a/libinstaller/linuxioctl.h
+++ b/libinstaller/linuxioctl.h
@@ -19,7 +19,7 @@

 #undef SECTOR_SIZE		/* Defined in msdos_fs.h for no good reason */
 #undef SECTOR_BITS
-#include <linux/ext2_fs.h>	/* EXT2_IOC_* */
+#include <ext2fs/ext2_fs.h>	/* EXT2_IOC_* */

 #ifndef FAT_IOCTL_GET_ATTRIBUTES
 # define FAT_IOCTL_GET_ATTRIBUTES	_IOR('r', 0x10, __u32)
diff --git a/win64/find-mingw64.sh b/win64/find-mingw64.sh
index c45db56..eadaab5 100755
--- a/win64/find-mingw64.sh
+++ b/win64/find-mingw64.sh
@@ -4,6 +4,7 @@ cc="$1"

 for prefix in \
     mingw64- \
+    x86_64-w64-mingw32- \
     x86_64-pc-mingw64- \
     x86_64-pc-mingw64msvc- \
     x86_64-pc-mingw32- \
--
1.9.1
